avm/ptn/authorization/role-definition/main.bicep

```diff
@@ -0,0 +1,106 @@
+metadata name = 'avm/ptn/authorization/role-definition'
+metadata description = 'This module deploys a custom role definition to a Management Group.'
+metadata owner = '@jtracey93'
+
+targetScope = 'managementGroup'
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. The location of the telemetry deployment to be created. Default is location of deployment.')
+param location string = deployment().location
+
+@description('Required. Array of custom role definitions to create on the management group.')
+param roleDefinition roleDefinitionType
+
+// ============== //
+// Resources      //
+// ============== //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.authorization-roledefinition.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  location: location
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource res_roleDefinition_mg 'Microsoft.Authorization/roleDefinitions@2022-05-01-preview' = {
```

Comment:
Is the plan to also add other scopes? Otherwise, nesting the MG-scope deployment may be redundant

Reply:
Yes there is a plan to add other scopes in the future, if requested, but for speed in ALZ we only need MG right now, so thats all ive developed so far

Reply:
ok gotcha, thanks.